merge.py

```diff
@@ -0,0 +1,188 @@
+    print('Command was {}, not a valid command.'.format(cmd))
+    return
+
+  if cmd == 'merge':
+    if cmt['user']['login'] not in AUTHORIZED_USERS:
+      post_error('Unauthorized users cannot merge: {}'.format(cmt['user']['login']))
+      print('Unauthorized user {} attempted to merge PR {}.'.format(cmt['user']['login'], pr_num))
+      return
+    # Kick off merge workflow
+    print('Command was merge, merging.')
+    if merge(pr_num):
+      post_info('Merge of PR#{} succeeded.', pr)
+      if not clean_up():
+        print("cleanup failed; dying.")
+        sys.exit(1)
+
+
+def merge(pr):
+  if not set_up():
+    post_error('Error setting up - please try again.', pr)
+    return False
+  # Make temp directory and cd into.
+  # Clone repository and configure.
+  print("Starting merge process for #{}.".format(pr))
+  clone_success = call(['git', 'clone', '-b', TARGET_BRANCH, 'https://github.com/{}/{}.git'.format(GITHUB_ORG, REPOSITORY), '/usr/local/google/home/jasonkuster/tmp/'], cwd='/usr/local/google/home/jasonkuster/tmp/')
+  if not clone_success == 0:
+    post_error('Couldn\'t clone from github/{}/{}. Please try again.'.format(GITHUB_ORG, REPOSITORY), pr)
+    return False
+  call(['git', 'remote', 'add', TARGET_REMOTE, 'https://git-wip-us.apache.org/repos/asf/{}.git'.format(REPOSITORY)], cwd='/usr/local/google/home/jasonkuster/tmp/')
+  call(['git', 'remote', 'rename', 'origin', SOURCE_REMOTE], cwd='/usr/local/google/home/jasonkuster/tmp/')
+  call('git config --local --add remote.' + SOURCE_REMOTE  + '.fetch "+refs/pull/*/head:refs/remotes/{}/pr/*"'.format(SOURCE_REMOTE), shell=True, cwd='/usr/local/google/home/jasonkuster/tmp/')
+  call(['git', 'fetch', '--all'], cwd='/usr/local/google/home/jasonkuster/tmp/')
+  print("Initial work complete.")
+  # Clean up fetch
+  initial_checkout = call(['git', 'checkout', '-b', 'finish-pr-{}'.format(pr), 'github/pr/{}'.format(pr)], cwd='/usr/local/google/home/jasonkuster/tmp/')
+  if not initial_checkout == 0:
+    post_error("Couldn't checkout code. Please try again.", pr)
+    return False
+  print("Checked out.")
+  # Rebase PR onto main.
+  rebase_success = call(['git', 'rebase', '{}/{}'.format(TARGET_REMOTE, TARGET_BRANCH)], cwd='/usr/local/google/home/jasonkuster/tmp/')
+  if not rebase_success == 0:
+    print(rebase_success)
+    post_error('Rebase was not successful. Please rebase against main and try again.', pr)
+    return False
+  print("Rebased")
+
+  # Check out target branch to here
+  checkout_success = call(['git', 'checkout', '{}/{}'.format(TARGET_REMOTE, TARGET_BRANCH)], cwd='/usr/local/google/home/jasonkuster/tmp/')
+  if not checkout_success == 0:
+    post_error('Error checking out target branch: master. Please try again.', pr)
+    return False
+  print("Checked out Apache master.")
+
+  # Merge
+  merge_success = call(['git', 'merge', '--no-ff', '-m', 'This closes #{}'.format(pr), 'finish-pr-{}'.format(pr)], cwd='/usr/local/google/home/jasonkuster/tmp/')
```

Comment:
I think this might need an autosquash so that fixup! and squash! commits are appropriately flattened?